File: Python/program/Ram.py
from Komponen import Komponen

class Ram(Komponen):
    def __init__(self, merk, nama, kapasitasGB, ddr):
        super().__init__(merk, nama)
        self.kapasitasGB = kapasitasGB
        self.ddr = ddr

    # Setter
    def setKapasitasGB(self, kapasitasGB):
        self.kapasitasGB = kapasitasGB

    def setDDR(self, ddr):
        self.ddr = ddr

    # Getter
    def getKapasitasGB(self):
        return self.kapasitasGB

    def getDDR(self):
        return self.ddr

    def displayInfo(self):
        print(f"{self.getNama()} ({self.kapasitasGB} GB, DDR: {self.ddr})")

File: Java/Program/Ram.java
// Ram.java
public class Ram extends Komponen {
    private int kapasitasGB;
    private int ddr;
    
    public Ram() {}
    
    public Ram(String merk, String nama, int kapasitasGB, int ddr) {
        super(merk, nama);
        this.kapasitasGB = kapasitasGB;
        this.ddr = ddr;
    }
    
    // Setter methods
    public void setKapasitasGB(int kapasitasGB) {this.kapasitasGB = kapasitasGB;}
    public void setDDR(int ddr) {this.ddr = ddr;}
    
    // Getter methods
    public int getKapasitasGB() {return kapasitasGB;}
    public int getDDR() {return ddr;}
    
    @Override
    public void displayInfo() {
        System.out.println(getNama() + " (" + kapasitasGB + " GB, ddr: " + ddr + ")");
    }
}